src/overlays/actors/ovl_En_Fr/z_en_fr.c

```diff
@@ -27,99 +183,1134 @@ const ActorInit En_Fr_InitVars = {
+static u8 sTimerJumpingOutOfWater[] = { 5,   // FROG_YELLOW
+                                        1,   // FROG_BLUE
+                                        7,   // FROG_RED
+                                        3,   // FROG_PURPLE
+                                        9 }; // FROG_WHITE
+
+// targetScale (default = 150.0) Actor scale target for Math_SmoothScaleMaxF
+// Used as a frog grows from hearing a new child song
+static f32 sGrowingScale[] = { 180.0,   // 1st incremental scale
+                               210.0,   // 2nd incremental scale
+                               240.0,   // 3rd incremental scale
+                               270.0 }; // Final scale
+
+static u8 sPitchSmallFrog[] = { 5,   // frog 0 (Yellow - C-Down Ocarina)
+                                2,   // frog 1 (Blue - A Button Ocarina)
+                                9,   // frog 2 (Red - C-Right Ocarina)
+                                11,   // frog 3 (Purple - C-Left Ocarina)
+                                14 }; // frog 4 (White - C Up Ocarina)
+
+static s8 sPitchLargeFrog[] = { -7,   // frog 0 (Yellow - C-Down Ocarina)
+                                -10,   // frog 1 (Blue - A Button Ocarina)
+                                -3,   // frog 2 (Red - C-Right Ocarina)
+                                -1,   // frog 3 (Purple - C-Left Ocarina)
+                                2 }; // frog 4 (White - C Up Ocarina)
+
+static u8 sJumpOrder[] = { FROG_BLUE,
+                           FROG_YELLOW,
+                           FROG_RED,
+                           FROG_PURPLE,
+                           FROG_WHITE,
+                           FROG_BLUE,
+                           FROG_YELLOW,
+                           FROG_RED };
+
+static u8 sOcarinaNotes[] = { 0,   // A Button
+                              1,   // C-Down
+                              2,   // C-Right
+                              3,   // C-Left
+                              4 }; // C-Up
+
+static s32 sDLists[] = { 0x060059A0, 0x06005BA0 };
+
+void EnFr_OrientUnderwater(EnFr* this) {
+    Vec3f vec1;
+    Vec3f vec2;
+    f32 xzDist;
+    s16 yaw;
+
+    vec1.y = 0.0f;
+    vec1.x = 0.0f;
+
+    xzDist = sLogSpotToFromWater[this->actor.params].xzDist;
+    this->xzDistToLogSpot = xzDist;
+    vec1.z = xzDist;
+
+    Matrix_RotateY(sLogSpotToFromWater[this->actor.params].yaw, MTXMODE_NEW);
+    Matrix_MultVec3f(&vec1, &vec2);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1AC0C.s")
+    this->actor.posRot.pos.x = this->posLogSpot.x + vec2.x;
+    this->actor.posRot.pos.z = this->posLogSpot.z + vec2.z;
+    this->actor.posRot.pos.y = sLogSpotToFromWater[this->actor.params].yDist + this->posLogSpot.y; // -80.0
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1AC34.s")
+    yaw = (s16)(sLogSpotToFromWater[this->actor.params].yaw * (32768.0f / M_PI)) + 32768;
 
+    this->actor.shape.rot.y = yaw;
+    this->actor.posRot.rot.y = yaw;
```

Comment:
you should be able to remove this temp by doing `posRot.rot.y = shape.rot.y = (calculation)`